subprojects/gradle/module-types/src/main/kotlin/com/avito/android/module_type/internal/CheckModuleDependenciesTask.kt

```diff
@@ -0,0 +1,103 @@
+package com.avito.android.module_type.internal
+
+import com.avito.android.Problem
+import com.avito.android.asRuntimeException
+import com.avito.android.module_type.DependencyRestriction
+import com.avito.android.module_type.ModuleTypeRootExtension
+import com.avito.android.module_type.ModuleWithType
+import com.avito.android.module_type.Severity
+import com.avito.logger.LoggerFactory
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.Directory
+import org.gradle.api.file.ProjectLayout
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.SetProperty
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.TaskAction
+import javax.inject.Inject
+
+internal abstract class CheckModuleDependenciesTask @Inject constructor(
+    projectLayout: ProjectLayout,
+) : DefaultTask() {
+
+    @get:Internal
+    val rootDir: Directory = projectLayout.projectDirectory
+
+    @get:Input
+    @get:Optional
+    abstract val severity: Property<Severity>
+
+    @get:Input
+    abstract val dependentProjects: SetProperty<String>
+
+    @get:Input
+    abstract val restrictions: ListProperty<DependencyRestriction>
+
+    @get:Internal
+    abstract val loggerFactory: Property<LoggerFactory>
+
+    @TaskAction
+    fun checkDependencies() {
+        val violations = ModulesRestrictionsFinder(readModules(), restrictions.get()).violations()
+
+        if (violations.isNotEmpty()) {
+            val logger = loggerFactory.get().create("")
```

Comment:
Didn't get how to create a logger for a task without additional tags.

Reply:
You have to specify a TAG, task name and project it's a metainfo
but TAG is a required thing

Reply:
I see. As we discussed, I've made a wrong assumption about default metadata.